nntrainer/src/neuralnet.cpp

```diff
@@ -249,6 +249,7 @@ int NeuralNetwork::init() {
     ml_logd("layer name: %s", l->getName().c_str());
 
   initialized = true;
+  setBatchSize(batch_size);
```

Comment:
batch size is to be decided at running time (training/testing)

Reply:
There is a scenario that a user is calling an equivalent consecutive function calls to alternate `training()/testing()` after `NN.init()` which Keras or other frameworks support to. 

In this scenario, because batch has not set (`NN.batchsize` has set but `setBatchSize` has not been called and it is not visible from API's point of view), it does not properly even it has been initiated. This patch is to alleviate the problem.

Reply:
Calling `setBatchSize()` at `init()` does not solve the problem but actually just hides it.
`setBatchSize()` must be called with each call to `training()/testing()` as different batch size can be used and is supported.

In the `nerualnet.cpp` I can see, `inference` and `train` both call `setBatchSize()` before running the model.
Validation and testing do not do this in the current implementation as their batchsize is not part of the interface yet.

Reply:
Do you have a suggestion? as long as we are opening other plumbing api's  (like NN->forward, getFlatgraph...) we should resolve this problem anyway.

 I had an alternative which didn't seem appealing:
1. create a `getBatchSize()`
2. call `setBatchSize(getBatchSize()` at user side.

Reply:
resolved offline.